### PR TITLE
Document `POST /build/.../_buildinfo` endpoint with OpenAPI

### PR DESCRIPTION
**Pull request by eduardoj**
### For reviewers

Access with this [link](https://obs-reviewlab.opensuse.org/eduardoj-apidocs_newpost_build_buildinfo/apidocs-new/index.html#/Build/post_build__project_name___repository_name___architecture_name___package_name___buildinfo) the documented endpoint in the review app.

### CODE REVIEW

File: src/api/public/apidocs-new/paths/build_project_name_repository_name_architecture_name_package_name_buildinfo.yaml [hunk 1]
```diff
@@ -33,3 +33,93 @@ get:
 
   tags:
     - Build
+
+post:
+  summary: Return build information about a specific package.
```
Comment (danidoni) [inline src/api/public/apidocs-new/paths/build_project_name_repository_name_architecture_name_package_name_buildinfo.yaml:38]:
```suggestion
  summary: Set the build information about a specific package.
```

**Comment by danidoni** [comment]
I believe you need to set `operationId: postBuildProjectRepositoryArchPackageBuildinfo`

File: src/api/public/apidocs-new/paths/build_project_name_repository_name_architecture_name_package_name_buildinfo.yaml (hunk 1 above)
Comment (eduardoj) [reply src/api/public/apidocs-new/paths/build_project_name_repository_name_architecture_name_package_name_buildinfo.yaml:38]: Nop. This endpoint doesn't set any build information.

File: src/api/public/apidocs-new/paths/build_project_name_repository_name_architecture_name_package_name_buildinfo.yaml [hunk 2]
```diff
@@ -33,3 +33,93 @@ get:
...
+  description: Get build information using the file passed in the request body.
```
[the remark this answers is not in the document]
Comment (eduardoj) [reply src/api/public/apidocs-new/paths/build_project_name_repository_name_architecture_name_package_name_buildinfo.yaml:39]: Again, this is not setting any information. As the description says, this endpoint gets build information using the file passed in the request body.

But I understand the confusion. Let me clarify this with another paragraph in the description.

[the remark this answers is not in the document]
Comment (eduardoj) [reply src/api/public/apidocs-new/paths/build_project_name_repository_name_architecture_name_package_name_buildinfo.yaml]: I added a sentence to make this more clear. Please take a look again.

**Comment by eduardoj** [comment]
> I believe you need to set `operationId: postBuildProjectRepositoryArchPackageBuildinfo`

This was not set for any of the endpoints documented, but the ones you documented. Why do you think this is needed?

**Comment by danidoni** [comment]
> This was not set for any of the endpoints documented, but the ones you documented. Why do you think this is needed?

I wasn't able to load this endpoint until I noticed that I had that id set in the get and I created it, but now that I think about it, it may be something else that made it work, because you are right, we didn't set that id in previous endpoints...

File: src/api/public/apidocs-new/paths/build_project_name_repository_name_architecture_name_package_name_buildinfo.yaml [hunk 3]
```diff
@@ -33,3 +33,96 @@ get:
 
   tags:
     - Build
+
+post:
+  summary: Return build information about a specific package.
```
Comment (hennevogel) [inline src/api/public/apidocs-new/paths/build_project_name_repository_name_architecture_name_package_name_buildinfo.yaml:38]:
```suggestion
  summary: Return build information a build description would produce
```

File: src/api/public/apidocs-new/paths/build_project_name_repository_name_architecture_name_package_name_buildinfo.yaml [hunk 4]
```diff
@@ -32,3 +32,96 @@ get:
...
+    - $ref: '../components/parameters/project_name.yaml'
+    - $ref: '../components/parameters/repository_name.yaml'
+    - $ref: '../components/parameters/architecture_name.yaml'
+    - in: path
+      name: package_name
+      schema:
+        type: string
+      required: true
+      description: |
+        Package name.
```
Comment (bmwiedemann) [inline src/api/public/apidocs-new/paths/build_project_name_repository_name_architecture_name_package_name_buildinfo.yaml:54]: Should we mention something about _multibuild (`osc -M foo`) that is probably appended as `:foo` here?

e.g. in openSUSE:Factory/avahi, a `osc -v build --noservice -M qt5 standard` calls
makeurl: https://api.opensuse.org ['build', 'openSUSE:Factory', 'standard', 'x86_64', 'avahi:qt5', '_buildinfo'] ['add=gdb', 'add=vim', 'add=strace', 'add=less']

That `add=` param would be useful to document, too.

Comment (eduardoj) [reply src/api/public/apidocs-new/paths/build_project_name_repository_name_architecture_name_package_name_buildinfo.yaml:54]: Just to make sure we are talking about the same API endpoint here (`POST /build/{project_name}/{repository_name}/{architecture_name}/{package_name}/_buildinfo`), could you run that `osc` command with the `-H` flag, to see what API calls are involved?

I fear that we are talking about another endpoint. It also could belong to the already documented [GET one](https://api.opensuse.org/apidocs-new/index#/Build/get_build__project_name___repository_name___architecture_name___package_name___buildinfo).

Comment (bmwiedemann) [reply src/api/public/apidocs-new/paths/build_project_name_repository_name_architecture_name_package_name_buildinfo.yaml:54]:
```
Getting buildinfo from server and store to /space/osc/openSUSE:Factory/avahi/.os
c/_buildinfo-standard-x86_64.xml
makeurl: https://api.opensuse.org ['build', 'openSUSE:Factory', 'standard', 'x86
_64', 'avahi:qt5', '_buildinfo'] ['add=gdb', 'add=vim', 'add=strace', 'add=less'
]


-- POST https://api.opensuse.org/build/openSUSE:Factory/standard/x86_64/avahi:qt
5/_buildinfo?add=gdb&add=vim&add=strace&add=less
send: b'POST /build/openSUSE:Factory/standard/x86_64/avahi:qt5/_buildinfo?add=gd
b&add=vim&add=strace&add=less HTTP/1.1\r\nAccept-Encoding: identity\r\nAccept: a
pplication/xml\r\nContent-type: application/octet-stream\r\nContent-length: 3319
8\r\nHost: api.opensuse.org\r\nUser-agent: osc/0.182.0\r\nConnection: close\r\n\
r\n'
send: b'#\n# spec file\n#\n# Copyright (c) 2023 SUSE LLC\n#\n# All modifications
 and additions to the file contributed by third parties\n# remain the property...
```

**Review by bmwiedemann — APPROVED** [review]

**Pull request merged**